### ISSUE TITLE
🐛 p.match is not a function

### ISSUE THREAD

**Issue by jhessin** [rome/tools#1538]
## Environment

```
$ rome rage --summary
Environment Variables
  Object {
    COLORTERM: "truecolor"
    TERM: "screen-256color"
    TERM_PROGRAM: "tmux"
    TMUX: "/private/tmp/tmux-501/default,13644,0"
  }

User Config
  unset

Rome Version
  10.0.4-beta

Node Version
  14.17.0

Platform
  darwin x64 20.3.0

Terminal Features
  Object {
    background: "unknown"
    colorDepth: 1
    columns: 100
    cursor: false
    hyperlinks: false
    isTTY: false
    unicode: false
  }

Client Flags
  Object {
    clientName: "cli"
    cwd: "/Users/jhessin/Projects/server-side-web-app-tutorial/sample_store"
    realCwd: "/Users/jhessin/Projects/server-side-web-app-tutorial/sample_store"
    silent: false
  }

Server Status
  Object {
    projects: Array [
      Object {id: 0}
      Object {id: 1}
    ]
    server: Object {
      heapTotal: 38_690_816
      pid: 99_664
      uptime: 0.63189209
    }
    workers: Array [
      Object {
        astCacheSize: 0
        heapTotal: 38_690_816
        ownedBytes: 0
        ownedFileCount: 0
        pid: 99_664
        uptime: 0.631804676
      }
    ]
  }

```

## Steps To Reproduce

1. Clone my repository [here](https://github.com/jhessin/server-side-web-app-tutorial)
2. Go to the `sample_store` directory.
3. run `yarn rome check --apply`

## Expected Results

The project is linted and fixes applied.

## Actual Results

The following error is produced:
```
 .pnp.js:5993:20 internalError/request  INTERNAL  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ p.match is not a function

   3. Function.resolveVirtual (.pnp.js:5993:20)
        5992 │   static resolveVirtual(p) {
      > 5993 │     const match = p.match(VIRTUAL_REGEXP);
             │                     ^
        5994 │     if (!match || !match[3] && match[5]) return p;
        5995 │     const target = ppath.dirname(match[1]);
   4. VirtualFS.mapToBase (.pnp.js:6030:21)
        6029 │   mapToBase(p) {
      > 6030 │     return VirtualFS.resolveVirtual(p);
             │                      ^
        6031 │   }
   5. VirtualFS.statPromise (.pnp.js:5771:40)
   6. PosixFS.statPromise (.pnp.js:5771:23)
   7. FileHandle.stat (.pnp.js:9003:26)
   8. stat (project-rome/@internal/core/server/fs/RecoveryStore.ts:480:32)
   9. Function.all
  10. ___R$project$rome$$internal$core$server$fs$RecoveryStore_ts$default.writeFiles (
      project-rome/@internal/core/server/fs/RecoveryStore.ts:450:3)
  11. ___R$project$rome$$internal$core$server$ServerRequest_ts$default.flushFiles (
      project-rome/@internal/core/server/ServerRequest.ts:311:2)
  12. ___R$$priv$project$rome$$internal$core$server$linter$Linter_ts$LintRunner.run (
      project-rome/@internal/core/server/linter/Linter.ts:448:21)
  13. ___R$$priv$project$rome$$internal$core$server$fs$glob_ts$GlobberWatcher.callback (
      project-rome/@internal/core/server/linter/Linter.ts:532:18)
  14. ___R$$priv$project$rome$$internal$core$server$fs$glob_ts$GlobberWatcher.flush (
      project-rome/@internal/core/server/fs/glob.ts:241:3)
  15. ___R$$priv$project$rome$$internal$core$server$fs$glob_ts$GlobberWatcher.init (
      project-rome/@internal/core/server/fs/glob.ts:296:3)
  16. ___R$project$rome$$internal$core$server$fs$glob_ts$Globber.watch (project-rome/@internal/core/server/fs/glob.ts:165:14
      )
  17. ___R$project$rome$$internal$core$server$linter$Linter_ts$default.runSingle (
      project-rome/@internal/core/server/linter/Linter.ts:592:21)
  18. ___R$project$rome$$internal$core$server$linter$Linter_ts$default.throwSingle (
      project-rome/@internal/core/server/linter/Linter.ts:630:20)
  19. Object.callback (project-rome/@internal/core/server/commands/check.ts:130:3)
  20. ___R$project$rome$$internal$core$server$Server_ts$default.dispatchRequest (
      project-rome/@internal/core/server/Server.ts:924:4)
  21. ___R$project$rome$$internal$core$server$Server_ts$default.handleRequest (
      project-rome/@internal/core/server/Server.ts:781:46)
  22. <anonymous> (project-rome/@internal/core/server/Server.ts:599:10)
  and 2 others...

  ⚠ This diagnostic was derived from an internal Rome error. Potential bug, please report if necessary.

━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

✖ Found 1 problem
```

**Comment by jhessin** [comment]
Okay back up... I first got interested in using Rome because of [this](https://www.sitepen.com/blog/rome-the-last-javascript-package) article about a new tool from the creator of yarn, babel, etc. But if this tool created by the same person who made yarn doesn't support yarn I'm baffled...

**Comment by ematipico** [comment]
Eventually Rome will be able to install dependencies too.

Not sure if it's worth investing resources on pnp.

**Comment by jamiebuilds** [comment]
Hi! Thank you for opening this issue

We recently made the decision to [migrate the Rome codebase to Rust](https://rome.tools/blog/2021/09/21/rome-will-be-rewritten-in-rust) and making some significant changes to its architecture. We've also decided not to release the previous version written in TypeScript.

We're going through and closing pull requests and issues that were specific to the TypeScript codebase, including this one. Thank you for taking the time to open it.

Thanks again!
